lib/spack/spack/schema/config.py

```diff
@@ -72,6 +72,7 @@
             "environments_root": {"type": "string"},
             "connect_timeout": {"type": "integer", "minimum": 0},
             "verify_ssl": {"type": "boolean"},
+            "custom_certs": {"type": "string"},
```

Comment:
I’d just call this `ssl_certs`

it would be nice if this could also be a directory via SSL_CERT_DIR, as I believe LLNL stores certs that way for its internal PKI.

I’m unsure what should happen if you set SSL_CERT_DIR and SSL_CERT_FILE, but I would default to whatever curl does.

Reply:
Yeah that is true. There is also `REQUEST_CA_BUNDLE` and `CURL_CA_BUNDLE` that I'm aware of, but I'm not sure what the hierarchy/relationship of these things are so I will have to do some digging.

Reply:
> We should make sure that this is translated over, if set, when spack runs curl:
>
>https://stackoverflow.com/questions/38571099/how-can-i-set-the-certificates-in-curl
>
>it would also be nice to respect SSL_CERT_DIR

@tgamblin when I read the docs it seems like this is a compile time option to configure curl to merge the certs.  At run time it seems like the supported option is to point to a file. https://curl.se/docs/sslcerts.html

If users set the `CURL_CA_BUNDLE` env variable then curl should pull the certificates from there without additional commands.

So I see 3 things we can do here:
1) if `SSL_CERT_FILE` is set then append the curl command with `--cacert [file]`
2) if `SSL_CERT_DIR` is set then modify the environment calling curl to include the `CURL_CA_BUNDLE` env variable
3) configure spack built curl to merge the certs using the flags from item 3 of the docs. I think this might be good to do in a separate PR for the curl package though.

Reply:
Hmm looks like I misread option 2. It is just a file as well, and `--certfile` is not supported for all versions of curl.  So looks like I just have option for curl, but for a file and not a directory.